soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,229 @@
+import robocup
+import play
+import behavior
+import skills.move
+import skills.capture
+import tactics.coordinated_pass
+import constants
+import main
+import enum
+
+
+## A testing play to demonstrate our ability to pass and recieve balls
+# One robot will pursue the ball while three other robots will pass the ball amongst themselves
+# they can only pass the ball at the corners and recieve at the corners and they cannot move
+# through the center of the square
+class FourCornerPass(play.Play):
+	class State(enum.Enum):
+		## One robot goes captures the ball while the other two gets on corners
+		# pursuing robot does not move on the first instance
+		setup = 1
+
+		## The robots pass to each other and begin setting up the next pass
+		# while the chasing robot goes for the ball.
+		passing = 2
+
+	def __init__(self):
+		super().__init__(continuous=True)
+
+		# register states - they're both substates of "running"
+		self.add_state(FourCornerPass.State.setup,
+					   behavior.Behavior.State.running)
+		self.add_state(FourCornerPass.State.passing,
+					   behavior.Behavior.State.running)
+
+		self.add_transition(behavior.Behavior.State.start,
+							FourCornerPass.State.setup, lambda: True,
+							'immediately')
+		#transition when ball is captured
+		self.add_transition(
+			FourCornerPass.State.setup, FourCornerPass.State.passing, lambda: self.is_setup_done() , 'all subbehaviors completed')
+
+		#transition when ball is passed or failed to pass
+		self.add_transition(
+			FourCornerPass.State.passing, FourCornerPass.State.setup, lambda: self.subbehavior_with_name('passer').state == behavior.Behavior.
+			State.completed or self.subbehavior_with_name('passer').state == behavior.Behavior.
+			State.failed, 'all subbehaviors completed')
+
+		#change this to adjust the square size
+		variable_square = 4.5
+
+		#NEVER CHANGE THIS
+		length = constants.Field.Length
+		width = constants.Field.Width
+
+		#if the square size is smaller than the width shrink it down otherwise
+		#max value is the width of the field.
+		if (variable_square < width) :
+			width = variable_square
+
+		#radius of robot
+		self.bot = constants.Robot.Radius
+		# the largest and smallest x and y postion possible for the square
+		self.max_x = (width / 2 - self.bot * 3)
+		self.min_x = -self.max_x
+		self.max_y = length/2 + self.max_x
+		self.min_y = length/2 - self.max_x
+
+		# the four courners
+		self.square_points = [
+			robocup.Point(self.max_x, self.max_y),
+			robocup.Point(self.min_x, self.max_y),
+			robocup.Point(self.min_x, self.min_y),
+			robocup.Point(self.max_x, self.min_y)]
+
+		# speed of the hunting robots
+		self.normal_speed = 2.0
```

Comment:
You should be able to read the specific config value from the config

Reply:
I meant this is more offense's speed want it to be editable too

Reply:
Looking over this again, I still think we should pull the max speed from the config for the offense robots and then do a percent of that for the defense. I would hate for us to tune down the speed to make things cleaner then forget to change this config value to be lower when we demo

Reply:
This is not possible at the current state of Python config values